homeassistant/components/wolflink/__init__.py

```diff
@@ -103,7 +104,9 @@ async def fetch_parameters(client: WolfClient, gateway_id: int, device_id: int):
     try:
         fetched_parameters = await client.fetch_parameters(gateway_id, device_id)
         return [param for param in fetched_parameters if param.name != "Reglertyp"]
-    except (ConnectError, ConnectTimeout) as exception:
-        raise UpdateFailed(f"Error communicating with API: {exception}") from exception
+    except (ConnectError, ConnectTimeout, FetchFailed) as exception:
+        raise ConfigEntryNotReady(
+            f"Error communicating with API: {exception}"
+        ) from exception
     except InvalidAuth as exception:
-        raise UpdateFailed("Invalid authentication during update") from exception
+        raise ConfigEntryNotReady("Invalid authentication during update") from exception
```

Comment:
The library should not raise InvalidAuth if there's a connection error. InvalidAuth should mean that the connection is ok, but the credentials are not ok.

Reply:
Library throws InvalidAuth only when WOLF cloud returns 403 during connection.
So you mean that, InvalidAuth shouldn't enter the 'retry flow', right? I mean if during startup we get the InvalidAuth it means that we have to change integration (or rather recreate).
The question is what kind of exception, I terms of correct HA behavior, should be thrown. UpdateFailed? InvalidAuth should be rethrow?

Reply:
Preferably we should start a new config flow to let the user authenticate again to correct the credentials. Search for `SOURCE_REAUTH` in the code base to see examples.

An MVP solution is to just fail the integration set up if credentials are incorrect and log an error message.

Reply:
Ok. So I will change it as it was earlier. For now. Since without this after updating HA I will have to setup integration again :). And I will deep into the SOURCE_REAUTH for future changes. Thank you for that I wasn't aware that I can do such things.